cmd/kube-apiserver/app/server.go

```diff
@@ -115,6 +116,9 @@ func Run(s *options.ServerRunOptions) error {
 	if err = s.Authentication.Apply(genericConfig); err != nil {
 		return fmt.Errorf("failed to configure authentication: %s", err)
 	}
+	if err := preflight.RunApiserverChecks(s); err != nil {
```

Comment:
It's not clear to me how this will make the resulting code any cleaner--or even work. WaitForAvailableEtcd is waiting on multiple go routines simultaneously, and signalling for all of them to stop when a verdict has been reached. Because utilwait.PollImmediate is polling one condition in one thread, I'm not sure what we would gain from it here. Could you please elaborate on how you picture it being used here?